lib/partial/transmuxer.js

```diff
@@ -0,0 +1,375 @@
+          type: 'audio',
+          data: data
+        });
+      });
+
+      pipeline.audioSegmentStream.on('done',
+                                     pipeline.trigger.bind(pipeline, 'done'));
+      pipeline.audioSegmentStream.on('partialdone',
+                                     pipeline.trigger.bind(pipeline, 'partialdone'));
+      pipeline.audioSegmentStream.on('endedtimeline',
+                                     pipeline.trigger.bind(pipeline, 'endedtimeline'));
+
+      pipeline.audioSegmentStream.on('timingInfo',
+                                     pipeline.trigger.bind(pipeline, 'audioTimingInfo'));
+
+      pipeline.adts
+        .pipe(pipeline.audioSegmentStream);
+    }
+
+    // emit pmt info
+    pipeline.trigger('trackinfo', {
+      hasAudio: !!pipeline.tracks.audio,
+      hasVideo: !!pipeline.tracks.video
+    });
+  });
+
+  pipeline.captionStream.on('data', function(caption) {
+    var timelineStartPts;
+
+    if (pipeline.tracks.video) {
+      timelineStartPts = pipeline.tracks.video.timelineStartInfo.pts || 0;
+    } else {
+      // This will only happen if we encounter caption packets before
+      // video data in a segment. This is an unusual/unlikely scenario,
+      // so we assume the timeline starts at zero for now.
+      timelineStartPts = 0;
+    }
+
+    // Translate caption PTS times into second offsets into the
+    // video timeline for the segment
+    caption.startTime = clock.metadataTsToSeconds(caption.startPts, timelineStartPts, options.keepOriginalTimestamps);
+    caption.endTime = clock.metadataTsToSeconds(caption.endPts, timelineStartPts, options.keepOriginalTimestamps);
+
+    pipeline.trigger('caption', caption);
+  });
+
+  pipeline = createPipeline(pipeline);
+
+  pipeline.metadataStream.on('data', pipeline.trigger.bind(pipeline, 'id3Frame'));
+
+  return pipeline;
+};
+
+var aacPipeline = function(options) {
+  var
+    pipeline = {
+    type: 'aac',
+    tracks: {
+      audio: {
+        timelineStartInfo: {
+          baseMediaDecodeTime: options.baseMediaDecodeTime
+        }
+      }
+    },
+    metadataStream: new m2ts.MetadataStream(),
+    aacStream: new AacStream(),
+    audioRollover: new m2ts.TimestampRolloverStream('audio'),
+    timedMetadataTimestampRolloverStream:
+      new m2ts.TimestampRolloverStream('timed-metadata'),
+    adtsStream: new AdtsStream(true)
+  };
+
+  // set up the parsing pipeline
+  pipeline.headOfPipeline = pipeline.aacStream;
+
+  pipeline.aacStream
+    .pipe(pipeline.audioRollover)
+    .pipe(pipeline.adtsStream);
+  pipeline.aacStream
+    .pipe(pipeline.timedMetadataTimestampRolloverStream)
+    .pipe(pipeline.metadataStream);
+
+  pipeline.metadataStream.on('timestamp', function(frame) {
+    pipeline.aacStream.setTimestamp(frame.timeStamp);
+  });
+
+  pipeline.aacStream.on('data', function(data) {
+    if (data.type !== 'timed-metadata' || pipeline.audioSegmentStream) {
+      return;
+    }
+
+    var audioTrack = {
```

Comment:
You should be using `pipeline.tracks.audio` here instead of creating a new variable. This will cause the `AudioSegmentStream` to be using a different track object than the transmuxer uses in `setBaseMediaDecodeTime`